### PR TITLE
feat(controllers): improve gateway controller configuration and code …

### PR DESCRIPTION
**Pull request by Tsukikage7**
### Summary

  This PR addresses the review comments and technical debt from PR #827 and #839, improving code
   quality and configurability of the gateway controller.

  ### Changes

  #### 1. Configurable Gateway Image and ImagePullPolicy

  Previously hardcoded values are now configurable:

  | Setting | Before | After (Default) |
  |---------|--------|-----------------|
  | Image | `mfordjody/pixiugateway:debug` | `apache/dubbo-go-pixiu:latest` |
  | ImagePullPolicy | `Always` | `IfNotPresent` |

  Configuration can be set via `GatewayConfig`:
  ```go
  config.ControllerConfig.Gateway.Image = "custom/image:tag"
  config.ControllerConfig.Gateway.ImagePullPolicy = "Always"

  2. Add Warning Log for Unsupported Filter Types

  Added logging in mergeFilterConfig default branch to help debug configuration issues:
  log.Printf("[WARN] mergeFilterConfig: unsupported filter type %q, replacing config instead of
  merging", filterType)

  3. Refactor Duplicate Policy Loading Code

  Merged duplicate implementations of LoadClusterPolicy and LoadClusterPolicyByClusterName into
  a shared private method loadClusterPolicyByName, reducing code duplication.

  4. Add Unit Tests

  New test coverage for:
  - config.go: GatewayConfig, TimeDuration, LeaderElection
  - utils_policy.go: splitClusterName edge cases

  Related Issues

  - Resolves review comments from PR #827
  - Resolves review comments from PR #839 (@No-SilverBullet)

  Test Plan

  - All existing tests pass
  - New unit tests pass (go test ./internal/controller/...)
  - Build succeeds (go build ./...)

  Checklist

  - Code follows project style guidelines
  - Added unit tests for new functionality
  - No breaking changes to public APIs

### CODE REVIEW

**Comment by Alanxtl** [comment]
our project use `import-formatter` to format import blocks, that's the reason why ur CI fails. For you, u should 
1. run `go install github.com/dubbogo/tools/cmd/imports-formatter@latest`
2. cd to the root dir of `dubbo-go-pixiu`
3. run `imports-formatter`

**Comment by Tsukikage7** [comment]
> our project use `import-formatter` to format import blocks, that's the reason why ur CI fails. For you, u should
> 
> 1. run `go install github.com/dubbogo/tools/cmd/imports-formatter@latest`
> 2. cd to the root dir of `dubbo-go-pixiu`
> 3. run `imports-formatter`

Thanks for the clarification. 

I will install and run `imports-formatter` from the project root to fix the import formatting and update this PR.

File: controllers/internal/controller/config/config.go [hunk 1]
```diff
@@ -33,8 +33,17 @@ const (
 	DefaultLogLevel         = "info"
 	DefaultMetricsAddr      = ":8080"
 	DefaultProbeAddr        = ":8081"
+	DefaultGatewayImage     = "apache/dubbo-go-pixiu:latest"
```
Comment (mfordjody) [inline controllers/internal/controller/config/config.go]: 这部分尽量镜像保持原来的先，因为方便我在集群里面的进行调试，always的目的也是如此，所以没有特地去编写单元测试，主要的目的是提升开发效率，这部分也会在后续会通过 helm 模板传值的方式去解决。

Comment (Tsukikage7) [reply controllers/internal/controller/config/config.go]: 好的，我改回去吧，一开始没太明白思路

**Comment by mfordjody** [comment]
LGTM.

**Comment by Tsukikage7** [comment]
@Alanxtl 帮忙re-run一下CI

**Review by Alanxtl — APPROVED** [review]
lgtm
